python/demo/demo_gmsh.py

```diff
@@ -183,15 +142,96 @@
             volume_entities.append(entity[1])
     model.addPhysicalGroup(3, volume_entities, tag=1)
     model.setPhysicalName(3, 1, "Mesh volume")
+    return model
+# -
 
-msh, mt, ft = gmshio.model_to_mesh(gmsh.model, mesh_comm, model_rank)
-msh.name = "hex_d2"
-mt.name = f"{msh.name}_cells"
-ft.name = f"{msh.name}_surface"
+# ## DOLFINx mesh creation and file output
+#
+# The following function creates a DOLFINx mesh from a Gmsh model, and
+# cell and facets tags. The mesh and the tags are written to an XDMF file
+# for visualisation, e.g. using ParaView.
+
+# +
 
-with XDMFFile(msh.comm, "out_gmsh/mesh.xdmf", "a") as file:
-    file.write_mesh(msh)
-    file.write_meshtags(mt, msh.geometry, geometry_xpath=f"/Xdmf/Domain/Grid[@Name='{msh.name}']/Geometry")
-    file.write_meshtags(ft, msh.geometry, geometry_xpath=f"/Xdmf/Domain/Grid[@Name='{msh.name}']/Geometry")
 
+def create_mesh(comm: MPI.Comm, model: gmsh.model, name: str, filename: str, mode: str):
```

Comment:
Maybe use pathlib instead of strings?

Reply:
I think for a demo `str` is fine - gives and indication without adding clutter.